src/Symfony/Component/PropertyAccess/PropertyAccessor.php

```diff
@@ -181,6 +181,7 @@ public function setValue(&$objectOrArray, $propertyPath, $value)
                     if ($propertyPath->isIndex($i)) {
                         if ($overwrite = !isset($zval[self::REF])) {
                             $ref = &$zval[self::REF];
+                            $ref = $zval[self::VALUE];
```

Comment:
why not change these lines to more readable `$zval[self::REF] = &$zval[self::VALUE]`?